tests/system/src/test/java/test/robot/javafx/stage/FocusParentWindowOnChildCloseTest.java

```diff
@@ -0,0 +1,159 @@
+    static Robot robot;
+    static Button button;
+    static Stage stage;
+    static Stage stage2;
+    static Scene scene;
+    static Alert alert;
+    static CountDownLatch startupLatch = new CountDownLatch(2);
+    static CountDownLatch alertShownLatch;
+    static CountDownLatch alertCloseLatch;
+
+    @Test(timeout = 15000)
+    public void focusRightParentOnChildWindowClose() throws Exception {
+        Thread.sleep(400);
+        clickButton();
+        fireOkInAlert();
+        Assert.assertTrue("Stage 1 should be focused", stage.isFocused());
+    }
+
+    private void clickButton() throws Exception {
+        alertShownLatch = new CountDownLatch(1);
+        mouseClick(button.getLayoutX() + button.getWidth() / 2, button.getLayoutY() + button.getHeight() / 2);
+
+        Thread.sleep(400);
+        waitForLatch(alertShownLatch, 10, "Failed to show Alert");
+    }
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        new Thread(() -> Application.launch(TestApp.class, (String[]) null)).start();
+        waitForLatch(startupLatch, 10, "FX runtime failed to start.");
+    }
+
+    @AfterClass
+    public static void exit() {
+        Platform.runLater(() -> {
+            stage.hide();
+            stage2.hide();
+        });
+        Platform.exit();
+    }
+
+    private void mouseClick(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + x),
+                    (int) (scene.getWindow().getY() + scene.getY() + y));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+    }
+
+    private static void fireOkInAlert() throws Exception {
+        alertCloseLatch = new CountDownLatch(1);
+        Platform.runLater(() -> {
+                    Button okButton = (Button) alert.getDialogPane().lookupButton(ButtonType.OK);
+                    System.out.println(okButton);
+                    okButton.fire();
+                });
+        Thread.sleep(400);
+        waitForLatch(alertCloseLatch, 10, "Failed to close alert.");
+    }
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) {
+            robot = new Robot();
+            stage = primaryStage;
+
+            createWindow1(stage);
+            createWindow2(stage);
+        }
+
+        private void createWindow1(Stage stage) {
+            stage.setTitle("Stage - 1");
+            button = new Button();
+            button.setText("Show Alert");
+            button.setOnAction(event -> showAlert(stage));
+
+            StackPane root = new StackPane();
+            root.getChildren().add(button);
+            scene = new Scene(root, 300, 250);
+            stage.setScene(scene);
+            stage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> Platform.runLater(startupLatch::countDown));
+            stage.show();
```

Comment:
The test may fail, if the stage does not get shown on foreground.
Please include this call : stage.setAlwaysOnTop(true);  before stage.show();

Reply:
Added. Did not change results.